sys/dev/etherswitch/e6000sw/e6000sw.c

```diff
@@ -252,15 +252,27 @@ e6000sw_probe(device_t dev)
 	    device_get_unit(sc->dev), "addr", &sc->sw_addr) != 0)
 		return (ENXIO);
 	if (resource_int_value(device_get_name(sc->dev),
-	    device_get_unit(sc->dev), "is6190", &is_6190) != 0)
+	    device_get_unit(sc->dev), "is6190", &is_6190) != 0) {
 		/*
 		 * Check "is8190" to keep backward compatibility with
 		 * older setups.
 		 */
 		resource_int_value(device_get_name(sc->dev),
 		    device_get_unit(sc->dev), "is8190", &is_6190);
+		/*
+		 * Check "is6190x" variant
+		 */
+		resource_int_value(device_get_name(sc->dev),
+		    device_get_unit(sc->dev), "is6190x", &is_6190x)
+		}
+	if (is_6190 !=0 && is_6190x != 0)
+	    if (bootverbose)
+	        device_printf(dev, "Cannot configure conflicting variants\n");
+	    return (ENXIO);
 	if (is_6190 != 0)
 		sc->swid = MV88E6190;
+	if (is_6190 != 0)
+		sc->swid = MV88E6190X;
```

Comment:
This part seems wrong
Because both can't be available (as it creates a conflict) but on the next line, it's checking if is_6190 != 0 two times, one time to set MV88E6190 and another time to set MV88E6190X. At the end it will set swid value to MV88E6190X, even if MV88E6190 (is_6190 != 0) is only available.